### PR TITLE
BUG/TST: verify an empty sequence file works gracefully

### PR DESCRIPTION
**Pull request by wasade**
I was getting the following under a parallel run:

``` python
Traceback (most recent call last):
  File "/home/mcdonadt/miniconda3/envs/deblur/bin/deblur", line 621, in <module>
    deblur_cmds()
  File "/home/mcdonadt/miniconda3/envs/deblur/lib/python3.5/site-packages/click/core.py", line 716, in __call__
    return self.main(*args, **kwargs)
  File "/home/mcdonadt/miniconda3/envs/deblur/lib/python3.5/site-packages/click/core.py", line 696, in main
    rv = self.invoke(ctx)
  File "/home/mcdonadt/miniconda3/envs/deblur/lib/python3.5/site-packages/click/core.py", line 1060, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/mcdonadt/miniconda3/envs/deblur/lib/python3.5/site-packages/click/core.py", line 889, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/mcdonadt/miniconda3/envs/deblur/lib/python3.5/site-packages/click/core.py", line 534, in invoke
    return callback(*args, **kwargs)
  File "/home/mcdonadt/miniconda3/envs/deblur/lib/python3.5/contextlib.py", line 77, in __exit__
    self.gen.throw(type, value, traceback)
  File "/home/mcdonadt/miniconda3/envs/deblur/lib/python3.5/site-packages/click/core.py", line 86, in augment_usage_errors
    yield
  File "/home/mcdonadt/miniconda3/envs/deblur/lib/python3.5/site-packages/click/core.py", line 534, in invoke
    return callback(*args, **kwargs)
  File "/home/mcdonadt/miniconda3/envs/deblur/bin/deblur", line 592, in workflow
    parallel_deblur(input_file_list, sys.argv, ref_db_fp, jobs_to_start)
  File "/home/mcdonadt/miniconda3/envs/deblur/lib/python3.5/site-packages/deblur/parallel_deblur.py", line 131, in parallel_deblur
    es))
RuntimeError: stdout:
stderr: Traceback (most recent call last):
  File "/home/mcdonadt/miniconda3/envs/deblur/bin/deblur", line 621, in <module>
    deblur_cmds()
  File "/home/mcdonadt/miniconda3/envs/deblur/lib/python3.5/site-packages/click/core.py", line 716, in __call__
    return self.main(*args, **kwargs)
  File "/home/mcdonadt/miniconda3/envs/deblur/lib/python3.5/site-packages/click/core.py", line 696, in main
    rv = self.invoke(ctx)
  File "/home/mcdonadt/miniconda3/envs/deblur/lib/python3.5/site-packages/click/core.py", line 1060, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/mcdonadt/miniconda3/envs/deblur/lib/python3.5/site-packages/click/core.py", line 889, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/mcdonadt/miniconda3/envs/deblur/lib/python3.5/site-packages/click/core.py", line 534, in invoke
    return callback(*args, **kwargs)
  File "/home/mcdonadt/miniconda3/envs/deblur/bin/deblur", line 584, in workflow
    negate=negate, threads_per_sample=threads_per_sample)
  File "/home/mcdonadt/miniconda3/envs/deblur/lib/python3.5/site-packages/deblur/workflow.py", line 688, in launch_workflow
    error_dist, indel_prob, indel_max)
  File "/home/mcdonadt/miniconda3/envs/deblur/lib/python3.5/site-packages/deblur/deblurring.py", line 111, in deblur
    seqs = get_sequences(input_seqs)
  File "/home/mcdonadt/miniconda3/envs/deblur/lib/python3.5/site-packages/deblur/deblurring.py", line 46, in get_sequences
    seqs = [Sequence(id, seq) for id, seq in input_seqs]
  File "/home/mcdonadt/miniconda3/envs/deblur/lib/python3.5/site-packages/deblur/deblurring.py", line 46, in <listcomp>
    seqs = [Sequence(id, seq) for id, seq in input_seqs]
  File "/home/mcdonadt/miniconda3/envs/deblur/lib/python3.5/site-packages/deblur/workflow.py", line 72, in sequence_generator
    raise skbio.io.FormatIdentificationWarning("input_fp does not appear "
skbio.io._warning.FormatIdentificationWarning: input_fp does not appear to be FASTA or FASTQ.

exit: 1
```

...seems like it was stemming from a warning raised by parsing an empty file. Unsure why a warning results in termination of the run but this resolves the problem

### CODE REVIEW

**Comment by antgonza** [comment]
👍

**Comment by mortonjt** [comment]
Did you run into this error when running real data?  

It looks like only a unittest is being added here.  Is this a fix?

**Comment by wasade** [comment]
Unsure why it doesn't trigger in test code, it triggers when running live. Might have to deal with how the parallel dispatch handles warnings which arise.

**Comment by wasade** [comment]
...anyway, pushed in the changes I made which allowed me to deblur a lot of things.

File: deblur/deblurring.py [hunk 1]
```diff
@@ -43,7 +43,10 @@ def get_sequences(input_seqs):
         If all the sequences do not have the same length either aligned or
         unaligned.
     """
-    seqs = [Sequence(id, seq) for id, seq in input_seqs]
+    try:
+        seqs = [Sequence(id, seq) for id, seq in input_seqs]
+    except:
+        seqs = []
```
Comment (mortonjt) [inline deblur/deblurring.py]: What do you think about print out a message in the log file, indicating that there were no sequences detected?

@amnona what do you think?

**Comment by mortonjt** [comment]
Hmm, looks like the try/except isn't being evaluated in the unittests.  Once that is added, I will merge this asap.

**Comment by wasade** [comment]
Warning in the seed comment on the PR actually stems from [here](https://github.com/biocore/deblur/blob/7224d40917f298d71ff76fa9ee3c981d9c68bada/deblur/workflow.py#L75). I do not know why the try/except works on real data but is not exercised here.

**Comment by wasade** [comment]
Just wanted to check, has this not been merged because of a 0.4% reduction in coverage?

**Comment by mortonjt** [comment]
I think it is good to go ...

**Pull request merged**